### PR TITLE
ci, doc: Add a link to Android APK CI artifact

### PR DESCRIPTION
**Pull request by hebasto**
Ability to download Android APK from CI was requested on the recent [call](https://github.com/BitcoinDesign/Meta/issues/185).

[Windows](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/win64/insecure_win_gui.zip?branch=pull/82)
[macOS](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/macos/insecure_mac_gui.zip?branch=pull/82)
[Android](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/android/insecure_android_apk.zip?branch=pull/82)

### CODE REVIEW

**Comment by hebasto** [comment]
@promag Could you suggest a better Android badge link?

**Review by promag — COMMENTED** [review]
Concept ACK.

Maybe change to:
https://img.shields.io/badge/OS-Windows-green
https://img.shields.io/badge/OS-macOS-green
https://img.shields.io/badge/OS-Android-green

**Comment by hebasto** [comment]
> Concept ACK.
> 
> Maybe change to: https://img.shields.io/badge/OS-Windows-green https://img.shields.io/badge/OS-macOS-green https://img.shields.io/badge/OS-Android-green

Thank you @promag! Done.

**Comment by hebasto** [comment]
@Bosch-0 

Is this what you were requested for?

File: .cirrus.yml [hunk 1]
```diff
@@ -320,3 +321,7 @@ task:
   env:
     << : *CIRRUS_EPHEMERAL_WORKER_TEMPLATE_ENV
     FILE_ENV: "./ci/test/00_setup_env_android.sh"
+  copy_artifacts_script:
+    - cp src/qt/android/build/outputs/apk/release/android-release-unsigned.apk ${CIRRUS_WORKING_DIR}/insecure_android.apk
+  insecure_android_apk_artifacts:
+    path: "insecure_android.apk"
```
Comment (promag) [inline .cirrus.yml:327]: nit, add newline.

**Review by promag — COMMENTED** [review]
It failed to install, but link works.

**Review by jarolrod — COMMENTED** [review]
Concept ACK

Can't test the resulting apk, but the link works

**Comment by hebasto** [comment]
> It failed to install, but link works.

Just leaving here a [link](https://github.com/bitcoin/bitcoin/pull/22469#issuecomment-881867614) to @icota's comment dated 2021-07-17.

**Comment by icota** [comment]
```
adb: failed to install insecure_android.apk: Failure [INSTALL_PARSE_FAILED_NO_CERTIFICATES: Failed collecting certificates for /data/app/vmdl895240733.tmp/base.apk: Package /data/app/vmdl895240733.tmp/base.apk has no certificates at entry AndroidManifest.xml]
```
Not sure why it worked before. Maybe `gradle` somehow found my dev cert from outside the container.

**Comment by hebasto** [comment]
@promag
> It failed to install, but link works.

Mind trying one more time?

File: .cirrus.yml (hunk 1 above)
Comment (hebasto) [reply .cirrus.yml]: Done.

**Comment by promag** [comment]
@hebasto same error.

**Comment by promag** [comment]
Tested ACK e9a1e98fa44547afcc9d284c38dc0442159abf7f

image

**Comment by hebasto** [comment]
Updated e9a1e98fa44547afcc9d284c38dc0442159abf7f -> 83f1893265bfa9b45dfc08cd2cf6d91c451e213d ([pr82.03](https://github.com/hebasto/gui-qml/commits/pr82.03) -> [pr82.04](https://github.com/hebasto/gui-qml/commits/pr82.04), [diff](https://github.com/hebasto/gui-qml/compare/pr82.03..pr82.04)):

- fixed filename in README.md

**Review by promag — COMMENTED** [review]
ACK 83f1893265bfa9b45dfc08cd2cf6d91c451e213d, just fixed url in readme.

**Review by shaavan — COMMENTED** [review]
Concept ACK

I was able to download the android apk through the link successfully:
```
https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/android/insecure_android_apk.zip?branch=pull/82
```

The app is working perfectly well on my phone (mi note 5 pro).

**Screenshot:**

WhatsApp Image 2021-11-12 at 4 23 11 PM

However, when I clicked on the link in the readme file:
```
https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/android/insecure_android_apk.zip
```

I received the following error page message:

Screenshot from 2021-11-12 12-56-52

I am not sure, but this seems like an unexpected thing and should be taken a look at.

**On a side note:**

I noticed a thing while installing the app on my phone. While installing, the shown icon of the app was a default package icon.

WhatsApp Image 2021-11-12 at 4 35 06 PM

But after installing the bitcoin core icon showed:

WhatsApp Image 2021-11-12 at 4 35 06 PM(1)

This bug could be taken up in a follow-up PR!

**Comment by promag** [comment]
@shaavan I think it will only work when this gets merged.

**Review by shaavan — APPROVED** [review]
> I think it will only work when this gets merged.

If that's the case, then I find no issue with this PR :)

ACK 83f1893265bfa9b45dfc08cd2cf6d91c451e213d

**Comment by shaavan** [comment]
> I think it will only work when this gets merged.

@promag @hebasto, the link (https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/android/insecure_android_apk.zip) is still not working.

**Comment by promag** [comment]
@shaavan lets wait master jobs finishes.

**Comment by promag** [comment]
@shaavan it works

**Pull request merged**